nixos/modules/services/security/esdm.nix

```
@@ -0,0 +1,138 @@
+{ lib, config, pkgs, ... }:
+
+let
+  cfg = config.services.esdm;
+  kernelVersion = config.boot.kernelPackages.kernel.version;
+  linux_6_3 = (lib.versionAtLeast kernelVersion "6.3");
+in
+{
+  options.services.esdm = {
+    enable = lib.mkEnableOption (lib.mdDoc "ESDM service configuration");
+    package = lib.mkPackageOptionMD pkgs "esdm" { };
+    kernelSupportEnable = lib.mkEnableOption (lib.mdDoc "Enable kernel support for the scheduler- and interrupt-entropy-source.");
+    serverEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Enable option for ESDM server service. If serverEnable == false, then the esdm-server
+        will not start. Also the subsequent services esdm-cuse-random, esdm-cuse-urandom
+        and esdm-proc will not start as these have the entry Want=esdm-server.service.
+      '';
+    };
+    cuseRandomEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Enable option for ESDM cuse-random service. Determines if the esdm-cuse-random.service
+        is started.
+      '';
+    };
+    cuseUrandomEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Enable option for ESDM cuse-urandom service. Determines if the esdm-cuse-urandom.service
+        is started.
+      '';
+    };
+    procEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Enable option for ESDM proc service. Determines if the esdm-proc.service
+        is started.
+      '';
+    };
+    verbose = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Enable verbose ExecStart for ESDM. If verbose == true, then the corresponding "ExecStart"
+        values of the 4 aforementioned services are overwritten with the option
+        for the highest verbosity.
+      '';
+    };
+    esdmHashName = lib.mkOption {
+      type = lib.types.str;
+      default = "sha3-512";
+      description = lib.mdDoc ''
+        Set the hash configration of the kernel module esdm-es.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable (
+    lib.mkMerge [
+      ({
+        systemd.packages = [ cfg.package ];
+
+        assertions = lib.lists.singleton {
+          assertion = linux_6_3 || cfg.kernelSupportEnable;
+          message = "Expected kernel-version >= 6.3. got kernel-version:${kernelVersion}";
```

Does this mean `kernelSupportEnable` is only required if you are using a kernel older than 6.3? If so it should be reflected in the description of the option.

It would also be nice if you add to the warning that ESDM expects a certain version.
```suggestion
          message = "ESDM expects kernel version >= 6.3. got kernel version:${kernelVersion}";
```